src/user_service/models.py

```diff
@@ -157,31 +166,56 @@ class UserStation(models.TextChoices):
         primary_key=True, default=uuid4, editable=False, unique=True
     )
     username = models.CharField(
-        max_length=255, unique=True, null=False, blank=False
+        max_length=128,
+        unique=True,
+        null=False,
+        blank=False,
+        validators=[
+            lambda username: UserModelValidator.username_validator(
+                username, min_length=4, max_length=128
+            ),
+        ],
+        help_text=_("Your unique username for CentaIMS")
     )
     email = models.EmailField(
         max_length=255,
         unique=True,
         blank=False,
         null=False,
     )
-    password = models.CharField(max_length=255, blank=False, null=False)
-    firstname = models.CharField(max_length=255)
+    password = models.CharField(
+        max_length=255,
+        blank=False,
+        null=False,
+        help_text=_(
+            "A secure password with at least one special character, uppercase letter, and number."
+        )
+    )
+    firstname = models.CharField(max_length=255, blank=False, null=False)
     middlename = models.CharField(max_length=255, blank=True)
     lastname = models.CharField(max_length=255, blank=False, null=False)
     role = models.OneToOneField(
-        "Role", on_delete=models.SET_NULL, null=True, blank=True
+        "Role",
+        on_delete=models.SET_NULL,
+        null=True,
+        blank=True,
+        related_name="user_role",
+        validators=[UserModelValidator.role_validator]
     )
     organisation = models.OneToOneField(
-        "Organisation", on_delete=models.CASCADE, null=True, blank=True
+        "Organisation", on_delete=models.CASCADE, null=True, blank=True, related_name="user_org"
```

Comment:
The default related name Django gives is `organisation` which I think is enough. This makes it simpler to have something like
```python
user = User.objects.get(id=1)
user.organization  # insetad of having to do user.user_org
```

What do you think?

Reply:
You'll have a problem with me on this project.... I write BRE you write AME. Massa